lib/ecto/uuid.ex

```diff
@@ -13,10 +13,28 @@ defmodule Ecto.UUID do
   @doc """
   Casts to UUID.
   """
-  def cast(<< a1, a2, a3, a4, a5, a6, a7, a8, ?-,
-              b1, b2, b3, b4, ?-,
-              c1, c2, c3, c4, ?-,
-              d1, d2, d3, d4, ?-,
+  def cast(<< a::64, ?-, b::32, ?-, c::32, ?-, d::32, ?-, e::96 >>) do
+    do_cast(<< a::64, b::32, c::32, d::32, e::96 >>)
```

Comment:
@josevalim Thanks for your feedback, that totally makes sense. Let me know if my update looks ok.